hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -567,6 +574,84 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 		log.Info("Reconciled userData Secret", "result", result)
 	}
 
+	// Validate tunedconfig input.
+	tunedConfig, err := r.getTunedConfig(ctx, nodePool, hcluster)
+	if err != nil {
+		setStatusCondition(&nodePool.Status.Conditions, hyperv1.NodePoolCondition{
+			Type:               hyperv1.NodePoolValidTunedConfigConditionType,
+			Status:             corev1.ConditionFalse,
+			Reason:             hyperv1.NodePoolValidationFailedConditionReason,
+			Message:            err.Error(),
+			ObservedGeneration: nodePool.Generation,
+		})
+		return ctrl.Result{}, fmt.Errorf("failed to get tunedConfig: %w", err)
+	}
+
+	setStatusCondition(&nodePool.Status.Conditions, hyperv1.NodePoolCondition{
+		Type:               hyperv1.NodePoolValidTunedConfigConditionType,
+		Status:             corev1.ConditionTrue,
+		Reason:             hyperv1.NodePoolAsExpectedConditionReason,
+		ObservedGeneration: nodePool.Generation,
+	})
+
+	// Check if config needs to be updated.
+	targetTunedConfigHash := hashStruct(tunedConfig)
+	isUpdatingTunedConfig := isUpdatingTunedConfig(nodePool, targetTunedConfigHash)
+	if isUpdatingTunedConfig {
+		setStatusCondition(&nodePool.Status.Conditions, hyperv1.NodePoolCondition{
+			Type:               hyperv1.NodePoolUpdatingTunedConfigConditionType,
+			Status:             corev1.ConditionTrue,
+			Reason:             hyperv1.NodePoolAsExpectedConditionReason,
+			Message:            fmt.Sprintf("Updating Tuned config in progress. Target Tuned config: %s", targetTunedConfigHash),
+			ObservedGeneration: nodePool.Generation,
+		})
+		log.Info("NodePool tuned config is updating",
+			"current", nodePool.GetAnnotations()[nodePoolAnnotationCurrentTunedConfig],
+			"target", targetTunedConfigHash)
+	} else {
+		removeStatusCondition(&nodePool.Status.Conditions, hyperv1.NodePoolUpdatingTunedConfigConditionType)
+	}
+
+	// 2. - Reconcile towards expected state of the world.
+	targetTunedConfigVersionHash := hashStruct(tunedConfig + targetVersion)
+	log.Info("Target TunedConfig Version Hash Info:",
+		"target hash", targetTunedConfigVersionHash, "version", targetVersion, "tunedConfig", tunedConfig)
+
+	// Tuned ConfigMaps exist for each NodePool config/version and follow "prefixName-configVersionHash" naming convention.
```

Comment:
I don't think this makes sense here, for .config is relevant because the whole payload including the OS is generated out of the .release + .config. In this case we just want to reconcile a config map that then the ntoperator watches.